src/main/java/com/twilio/http/Request.java

```diff
@@ -175,9 +176,19 @@ private String buildURL() {
                 host = joinIgnoreNull(".", product, targetEdge, targetRegion, domain);
             }
 
-            return new URI(parsedUrl.getProtocol(), parsedUrl.getUserInfo(), host, parsedUrl.getPort(),
-                    parsedUrl.getPath(), parsedUrl.getQuery(), parsedUrl.getRef()).toURL().toString();
-        } catch (final MalformedURLException | URISyntaxException e) {
+            String urlPort = null;
+            if (parsedUrl.getPort() != -1) {
+                urlPort = ":" + parsedUrl.getPort();
+            }
+            String protocol = parsedUrl.getProtocol() + "://";
+            String[] pathPieces = parsedUrl.getPath().split("/", 0);
+            pathPieces[pathPieces.length-1] = URLEncoder.encode(pathPieces[pathPieces.length-1], "UTF-8");
```

Comment:
I don't understand this `pathPieces` logic. Needs a comment.

Reply:
Comment added. The idea is to only encode the path portion of the URL.

Reply:
So if you have `/foo|bar/baz|qux` only `baz/qux` gets encoded?

Reply:
Yes, `baz|qux` would only be encoded. Based on our tests, I'm assuming this case does not happen. If we do expect that use case, then perhaps we should encode all pieces of the path, save `/` characters.

Reply:
We don't control where the characters needing encoding would be. Best to encode them all.

Reply:
Updated